src/main/java/mate.academy/lib/Injector.java

```diff
@@ -1,13 +1,115 @@
+    private static final String CLASS_EXTENSION = ".class";
+    private final Map<Class<?>, Object> instances = new HashMap<>();
+    private final Map<Class<?>, Class<?>> implementations;
+
+    private Injector() {
+        List<Class<?>> interfaces = findInterfaces();
+        implementations = findImplementations(interfaces);
+    }
 
     public static Injector getInjector() {
         return injector;
     }
 
     public Object getInstance(Class<?> interfaceClazz) {
-        return null;
+        Class<?> implementationClazz = implementations.get(interfaceClazz);
+        boolean isPresent = instances.containsKey(implementationClazz);
+        if (isPresent) {
+            return instances.get(implementationClazz);
+        }
+        if (implementationClazz != null) {
+            try {
+                Object o = implementationClazz.getDeclaredConstructor().newInstance();
+                Field[] fields = o.getClass().getDeclaredFields();
+                for (Field field : fields) {
+                    Annotation[] annotations = field.getAnnotations();
+                    if (isAnnotationPresent(annotations, Inject.class)) {
+                        field.setAccessible(true);
+                        field.set(o, getInstance(field.getType()));
+                        field.setAccessible(false);
+                    }
+                }
+                instances.put(implementationClazz, o);
+                return o;
+
+            } catch (ReflectiveOperationException e) {
+                throw new RuntimeException("Something went wrong during object creation.");
+            }
+        }
+        throw new RuntimeException(
+                "There is missing implementation with @Component annotation for class"
+                        + interfaceClazz.getName());
+    }
+
+    private boolean isAnnotationPresent(Annotation[] annotations,
+            Class<? extends Annotation> annotationClass) {
+        return Arrays.stream(annotations)
+                .anyMatch(annotation -> annotation.annotationType().equals(annotationClass));
+    }
+
+    private List<Class<?>> findClassesInPackage(String packageName) {
+        ClassLoader classLoader = ClassLoader.getSystemClassLoader();
+        InputStream stream = classLoader.getResourceAsStream(packageName);
+        return new BufferedReader(new InputStreamReader(stream)).lines()
+                .filter(i -> i.endsWith(CLASS_EXTENSION))
+                .map(i -> i.substring(0, i.indexOf(CLASS_EXTENSION)))
+                .map(i -> getClazz(packageName, i)).collect(Collectors.toList());
+    }
+
+    private Class<?> getClazz(String packageName, String name) {
+        try {
+            return Class.forName(
+                    packageName.replace(PATH_DELIMITER, CLASSNAME_DELIMITER) + "." + name);
+        } catch (ClassNotFoundException e) {
+            throw new RuntimeException("There is no such class in your project.");
```

Comment:
The error message in this `RuntimeException` is too generic. Consider including the class name that was not found to provide more context for debugging.